### ISSUE TITLE
Reveal Modal

### ISSUE THREAD

**Issue by rickhumphries** [buttercupstraining/lrnwebcomponents#24]
Might need to have a think about how much we want to abstract this to make it generic (as it's essentially just a modal!) but we have a button with a cartoon pharmacist in it on the pollen theme which then reveals a modal:

![image](https://user-images.githubusercontent.com/4973810/107589399-c9b65600-6bfd-11eb-8009-4b6493121267.png)

![image](https://user-images.githubusercontent.com/4973810/107589417-daff6280-6bfd-11eb-881b-9bf49c8ccee2.png)

Shoelace element for this is 'dialog' - https://shoelace.style/components/dialog

**Comment by BethQuibell** [comment]
There are also some variations on this one depending on whether the info is relevant to a community pharmacy, a dispensing doctors, or a hospital pharmacy.

Community:
![image](https://user-images.githubusercontent.com/64269591/108194154-2dfd7c00-710e-11eb-851c-61a51952947f.png)
![image](https://user-images.githubusercontent.com/64269591/108194201-3c4b9800-710e-11eb-8025-2f34b7844289.png)

Dispensing Doctors:
![image](https://user-images.githubusercontent.com/64269591/108193848-cfd09900-710d-11eb-8bb5-fd6a5eded51d.png)
![image](https://user-images.githubusercontent.com/64269591/108193894-de1eb500-710d-11eb-9ee3-514433c635f7.png)

Hospital:
![image](https://user-images.githubusercontent.com/64269591/108193947-f098ee80-710d-11eb-95d9-d1c930e27efd.png)
![image](https://user-images.githubusercontent.com/64269591/108193993-fdb5dd80-710d-11eb-807f-a9dd739693d5.png)

**Comment by rickhumphries** [comment]
Good spot @BethQuibell, thanks! 

@JackWillDavis - happy to use any modal solution you'd like for this. This is the shoelace one - https://shoelace.style/components/dialog

In terms of the component itself, we don't need to make this generic. I would treat it in a similar way to how you've done the GPhC Standards Box and just give the author the choice between 'Default, Community, Hospital and Dispensing Doctors'. Then the person image and colours would just update based on that one choice. 

Could name the element `<bc-info-popup>`

**Comment by JackWillDavis** [comment]
Yeah if possible please. Just managed to get shoelace in and configured so just need to lay it all out now.

**Comment by rickhumphries** [comment]
Awesome! Yeh all the characters should be here now:
\\192.168.1.6\Public Share\DEPARTMENTS\Development Team\Design Files\Course Design\LEVEL 2\Character\svg files
\\192.168.1.6\Public Share\DEPARTMENTS\Development Team\Design Files\Course Design\LEVEL 2\Character Design\svg files

**Comment by JackWillDavis** [comment]
@rickhumphries Nice one thanks, I'll grab them now.

Default: https://courses.buttercupstraining.co.uk/l2psvac/node/101

@BethQuibell - would it be possible to get links to those you found please? Just so I can grab all of the colours etc.

**Comment by BethQuibell** [comment]
@JackWillDavis - yep, they're all on this page https://courses.buttercupstraining.co.uk/da/node/200

**Comment by JackWillDavis** [comment]
Brill, thanks a lot!

**Comment by rickhumphries** [comment]
@JackWillDavis one thing I'd forgotten about functionality-wise with these modals is how they react for mobile devices. If you reduce the browser window size you'll see they change from being modals to more like simple boxes so we'll need some media queries on the component itself to take that into consideration:

![image](https://user-images.githubusercontent.com/4973810/108518338-588f3680-72c0-11eb-85db-13e345c631bf.png)

**Comment by JackWillDavis** [comment]
Yeah that should be fine, I'll just duplicate the slotted content and have one show on different screen sizes.

**Comment by JackWillDavis** [comment]
Got most of this set up now, just needs a few tweaks like positioning of the dialog and characters etc.

![image](https://user-images.githubusercontent.com/2301874/108530211-41a31100-72cd-11eb-931f-884b98717afc.png)

Mobile view will display differently (without popup functionality).

![image](https://user-images.githubusercontent.com/2301874/108530248-4bc50f80-72cd-11eb-8052-5114a81ef784.png)

The only major thing I'm coming across is duplicating the slotted content as it is only processed the once (which is why there is no "This is bc-info-popup" text on the mobile view. Usually you would name slots which are then placed in different places but just thinking about how we would do that in a way which didn't make it too cumbersome for the admin adding the content.

Current options are:

* To have the mobile view show less text which we can set in a property (instead of a slot)
* To write some JS which grabs the innerHTML of the desktop modal and prints it out in mobile view

HAX is all wired in now in any case so I'll just do a little more and then get it in place on the dev server. It's also displaying the SVG in an img tag but I'll switch that to `simple-icon` on the dev server before finishing it off.

**Comment by rickhumphries** [comment]
Looks great @JackWillDavis ! Glad the shoelace library hasn't thrown up any major issues :)

Yeh tricky one if you can't have the slotted content be replicated in two different places. I'm less in favour of the mobile view showing less text to be honest as it'd be good to just know that the content added by the author will show in it's entirety across all screen sizes - so the JS option would be preferred. Sounds like a good topic for Wednesday anyway!

**Comment by JackWillDavis** [comment]
This is all set up, including replicating slotted content to the mobile view and it's working locally. 

I'm going to have to wait to import it into ELMSLN as it requires Shoelace/Stencil node_modules and they all need to have dependencies included along with dependency links rewritten so I'll need to talk to Bryan about it before continuing. I'll send him a message but it might be something we could discuss in the weekly if not resolved by then?
